### DISCUSSION TITLE
Help with API: I can reply to my own toots, but not toots by other accounts

### DISCUSSION THREAD

**Discussion by ShredderWalnut** [mastodon/mastodon#17574] [Q&A] [answered]
Hello,

I am playing with the API in JavaScript for fun and want to reply to another toot via the API. Here’s part of the HTTP request:

```language-json
{
	"url": endpoint,
	"method": "POST",
	"data": {
		"status": toot,
		"visibility": “public”,
		"in_reply_to_id": ID,
		"spoiler_text": contentWarning
		},
	"headers": {
		"Authorization" : "Bearer " + token,
		"idempotency_key" : id_key
		}
}
```
I’m having trouble with the `in_reply_to_id` key, whose `ID` value is the ID of the toot I’m trying to reply to. I discovered that using this method, apparently I could only reply to my own toots. When I attempt to reply to other public toots by accounts that I follow, it all failed.

I read the [relevant section](https://docs.joinmastodon.org/methods/statuses/) in the docs and didn’t understand the issue. Apologies for the noob question and thanks in advance! <3

**Comment by ClearlyClaire** [comment, marked as answer]
I see nothing wrong in what you're showing, but maybe you are using an incorrect ID? How are you determining the ID of the toot to be replied?

**Reply by ShredderWalnut** [reply]
Thank you for your reply!

I determine the ID by copying the toot URL and using the string of numbers at the end.

My account is on mastodon.social and this method works. The failed tests were trying to reply to toots from botsin.space.

**Reply by ClearlyClaire** [reply]
I thought so. The number in a toot's URL is only the toot ID on the Mastodon server it was authored on, the toot ID is different on your server.

If you look at a remote toot from mastodon.social's web interface, you will notice that the number is different in the web interface URL than on the toot's home server. The number you need to use here is the one that shows up in mastodon.social's web interface.

**Reply by ShredderWalnut** [reply]
Ah! Thank you so much for solving the mystery for me. I really appreciate your help.

**Reply by ShredderWalnut** [reply]
Hi @ClearlyClaire , I have a follow-up question and hope you don't mind answering.

Is it possible to programmatically retrieve the "correct" (i.e. as appears on my Mastodon instance) toot ID from the original (i.e. their Mastodon instance) URL? For example, given a URL like `https://botsin.space/@wikipediahaiku/107807684574492607`, how to programmatically retrieve the ID at the end of `https://mastodon.social/web/@wikipediahaiku@botsin.space/107807684947855970`?

Thank you for your patience and time!

**Reply by ClearlyClaire** [reply]
Yes, use `/api/v2/search`: https://docs.joinmastodon.org/methods/search/

**Reply by ShredderWalnut** [reply]
Thank you so much!
